### PR TITLE
feat(CRO): CW-21638 update cronos to v2

### PR DESCRIPTION
**Pull request by hankliu418**
----

*Checklist:*

- README.md includes:
  - [ ] The details of signing data and transaction data, this includes parameters.
  - [ ] Official docs or white paper.
  - [ ] Website or api can query assets and broadcast transaction.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

 
 
 
 
 
 **PR Summary by Typo**
------------

**Overview**
This PR updates the Cronos SDK to v2, including dependency updates, script and signature formatting changes, and the addition of unit tests.  It also refactors transaction utility functions and updates the `package.json` with new scripts.

**Key Changes**
- Updates `@coolwallet/core` dependency to `2.0.0-beta.20`.
- Modifies scripts and signatures in `params.ts` to use template literals.
- Updates transaction utility functions to use the latest `@coolwallet/core` methods.
- Adds unit tests for address generation and transaction signing.
- Introduces new npm scripts for testing.

**Recommendations**
Not deployment ready. While the unit tests are a valuable addition, they currently lack sufficient coverage to ensure production readiness. Expand test cases to cover various scenarios, including edge cases and error handling, within the `index.spec.ts` file.  Additionally, consider adding integration tests to validate interactions with the updated `@coolwallet/core` dependency.  Finally, update the version in `package.json` to reflect that this is a beta release, e.g. `2.0.0-beta.1`.
 

 <h6>To turn off PR summary, please visit <a href="https://app.typoapp.io/settings/notification?tab=codeHealth">Notification settings</a>.</h6>

### CODE REVIEW

**Comment by cbx-tech** [comment]
Task linked: [CW-21638 CRONOS](https://app.clickup.com/t/25577573/CW-21638)

File: packages/coin-cro/tests/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    if (cardType === CardType.Lite) {
+      transport = (await createTransport('http://localhost:9527', CardType.Lite))!;
+    } else {
+      transport = (await createTransport())!;
+    }
+    props = await initialize(transport, mnemonic);
+  });
+
+  describe('Test Get Address', () => {
+    it('ETH test get address 0', async () => {
```
Comment (gundomlegend) [inline packages/coin-cro/tests/index.spec.ts:34]: 應該是 CRO

File: packages/coin-cro/tests/index.spec.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      const address = await croSDK.getAddress(transport, props.appPrivateKey, props.appId, 0);
+      const expectedAddress = 'cro1y2lk8ex0geseke743t7snntkw9aql090x8p4d6';
+      expect(address.toLowerCase()).toEqual(expectedAddress.toLowerCase());
+    });
+  });
+
+  describe('Sign Transfer Tx', () => {
+    it('ETH test sign transaction', async () => {
```
Comment (gundomlegend) [inline packages/coin-cro/tests/index.spec.ts:42]: 應該是 CRO

File: packages/coin-cro/tests/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { CardType, Transport } from '@coolwallet/core';
+import { initialize, getTxDetail, DisplayBuilder } from '@coolwallet/testing-library';
+import isEmpty from 'lodash/isEmpty';
+import { createTransport } from '@coolwallet/transport-jre-http';
+import * as utils from 'web3-utils';
+import CRO, { TX_TYPE } from '../src';
+import { CHAIN_ID, MsgSend, SignMsgSendType } from '../src/config/types';
+// import { signTx } from '../src/config/types';
```
Comment (gundomlegend) [inline packages/coin-cro/tests/index.spec.ts:8]: 好像可以移除

**Review by gundomlegend — APPROVED** [review]
